proposals/infra/INF-0005-Long-Vector-TestPlan.md

```diff
@@ -0,0 +1,199 @@
+    - TBD: Where do we test these? Tex mentioned some existing tests but it sounds like we don't want to use those. And they aren't HLK tests anyways.
+    - TODO: Discuss how we want to test ops that don't boil down to a single DXIL op. Still seems like something we should test in the HLK?
+
+3. Standard loading and storing of long vectors
+    - TODO: Do we have tests that do this today?
+
+# Vector Sizes to test
+
+I don't think there are any particularily interesting vector sizes to test. So I propose testing sizes of 5, 25, 100, 500, 
+
+# Phases
+Do the test work in two simple phases.
+
+1. Implement and validate (locally against WARP) the 3 test categories.
+2. HLK related work:
+    - Add a SM 6.9 HLK requirement. Includes updating the HLK requirements doc.
+    - Update mm_annotate_shader_op_arith_table.py to annotate the new test cases with HLK GUIDS and requirements
+    - Add new tests to HLK playlist
+
+# New HLK Tests
+mm_annotate_shader_op_arith_table.py (WinTools repo) will need to be updated to recognize any new tests (not test cases) added. And additional GUIDs addded for new test cases. mm_annotate_shader_op_arith_table.py is called by mm-hlk-update.py when converiting from 'ExecTests' to the HLK 'DxilConf' tests. The aformentioned *table.py script is run by Integration.HLKTestsUpdate.yaml
+
+# Test Validation Requirements
+The following statements must be true and validated for this work to be considered completed.
+ - All new test cases pass when run locally against a WARP device
+ - All new test cases are confirmed to be lit up and available in the HLK when the target device reports support
+ - All new tests/test cases are added to the HLK playlist
+
+# Requirements
+    - Gregs long vector changes. Check-in ETA of 3/7/25.
+    - WARP long vector support (Jesse). Needs Gregs work, or a private WIP branch of Gregs work. ETA of ~1 week to implement.
+
+# Notes
+    - The 'ExecTests' (ExecutionTest.cpp) are used to generate the HLK tests (DxilConf tests)
+
+# Open Questions
+    1. How does the HLK work for IHVs running DXIL tests given that we're undocked. Or does that not matter because the DXIL HLK tests provide raw DXIL? Just trying to understand the HLK testing lifecycle here vs what I was used to in the OS.
+    2. Do we need a new HLK requirement? If so, we probably need to get that in soon. That could also require HLK gatherer updates.
+        - For tests where we only add new test cases we do not need to update HLK test GUIDs.
+        - If we want to add a new HLK requirement, then we will probably want to create new tests which consume that requirement. Individual test cases can not have individual HLK requirements.
+        - The existing HLK reqs we have look pretty dated. "Device.Graphics.WDDM27.AdapterRender.D3D12.DXILCore.ShaderModel65.CoreRequirement". Add a requirement for SM 6.9? We're requiring Long Vectort support for SM 6.9
+    3. Do the work in two 'phases'? Add the tests. Then do the HLK side.
```

Comment:
I think we should build up a plan that involves multiple deliveries to the IHVs.  If we can identify a set of high-priority tests (eg those needed by a particular demo) we could try and get those out first to support IHVs building drivers for the preview release.

Reply:
I was thinking about this and if IHVs are OK with it I think the easiest way is to just share stand-alone test binaries/collateral. Can you help me sync up with the right people to figure out what works for IHVs? What have we done in the past?

Reply:
Yup, we can sort that out, but I'm more interested in what and when we'll deliver these rather than the mechanism used to do it.

Reply:
Actually, given that the DXC repo is public we should be able to just share build/run instructions for the tests once checked in?

Reply:
Let's discuss this further offline. I'll make a note to ask you.